src/mockServiceWorker.js

```diff
@@ -115,26 +115,29 @@ async function handleRequest(event, requestId) {
   // Ensure MSW is active and ready to handle the message, otherwise
   // this message will pend indefinitely.
   if (activeClientIds.has(client.id)) {
-    const clonedResponse = response.clone()
-
-    sendToClient(client, {
-      type: 'RESPONSE',
-      payload: {
-        requestId,
-        type: clonedResponse.type,
-        ok: clonedResponse.ok,
-        status: clonedResponse.status,
-        statusText: clonedResponse.statusText,
-        body: clonedResponse.body === null ? null : await clonedResponse.text(),
-        headers: serializeHeaders(clonedResponse.headers),
-        redirected: clonedResponse.redirected,
-      },
-    })
+    processResponse(client, requestId, response)
   }
 
   return response
 }
 
+async function processResponse(client, requestId, response) {
```

Comment:
Looks like this only moves the block outside. Is there a reason to do so?

We should be cautious when it comes to the service worker changes, as each such change is a breaking change. I'd say we can keep the worker file as-is without this change, it should produce the same result.

Reply:
in this way we are not awaiting the `.text` when handling the response

Reply:
Could we achieve the same by wrapping it in a self-invoking function?

```js
if (activeClientIds.has(client.id) {
  (async function handleResponse() {
    const clonedResponse = response.clone()
    // ....
  })()
}

It'd be useful to comment next to that function why it's structured that way (that awaiting on `.text` may cause the response handling to timeout). 
```

Reply:
Yes we will have the same result